integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,178 @@
+import logging
+
+from django.contrib import messages
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...decorators import permission_required
+from ...forms import (
+    ContactForm,
+    LocationForm,
+    OfferForm,
+    OfferVersionForm,
+    OrganizationForm,
+)
+from ...models import Offer, User
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.edit_offer"), name="dispatch")
+class OfferFormView(TemplateView):
```

Comment:
```suggestion
class OfferFormView(SingleObjectMixin, TemplateView):
```
You could let this view inherit from [SingleObjectMixin](https://docs.djangoproject.com/en/4.2/ref/class-based-views/mixins-single-object/#singleobjectmixin) and if you want to make use of the inbuilt `get_object()` method to retrieve the current offer instance. This works best if the url parameter is renamed from `offer_id` to just `pk`.

Reply:
If I'm not missing anything, this actually makes the code more verbose - if no pk or slug are passed in the url, `get_object()` throws an error. Is it worth using and catching this?

Reply:
Ah ok, right, we'd need a base class for both the update & create views (e.g. `OfferFormMixin`), and then two views which inherit from this (e.g. `CreateOfferView` with `self.object = None` and `UpdateOfferView` with `self.object = self.get_object()`) and instantiate the corresponding view in the URL config, at least this is how Django is doing it:
https://github.com/django/django/blob/c3862735cd8c268e99fb8d54c3955aacc4f2dc25/django/views/generic/edit.py#L169-L206

Could also be more work than necessary, so I'm also fine with keeping it the way it is.
I'd just prefer if we could throw an error if an invalid id is given (at the moment, this is equivalent to creating a new offer).